pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -138,11 +142,81 @@ def __init__(
-
         self.replace_sampler_ddp = replace_sampler_ddp
 
+    def handle_given_plugins(self, plugins: Optional[Sequence]):
+        plugins = plugins if plugins is not None else []
+
+        if isinstance(plugins, str):
+            plugins = [plugins]
+
+        if not isinstance(plugins, Sequence):
+            plugins = [plugins]
+
+        training_type = None
+        precision = None
+        cluster_environment = None
+
+        for plug in plugins:
+            if isinstance(plug, str):
+                self.set_distributed_mode(plug)
+
+            elif isinstance(plug, TrainingTypePlugin):
+                if training_type is None:
+                    training_type = plug
+
+                else:
+                    raise MisconfigurationException(
+                        'You can only specify one precision and one training type plugin. '
+                        'Found more than 1 training type plugin'
+                    )
+            elif isinstance(plug, PrecisionPlugin):
+                if precision is None:
+                    precision = plug
+                else:
+                    raise MisconfigurationException(
+                        'You can only specify one precision and one training type plugin. '
+                        'Found more than 1 precision plugin'
+                    )
+
+            elif isinstance(plug, ClusterEnvironment):
+                if cluster_environment is None:
+                    cluster_environment = plug
+                else:
+                    raise MisconfigurationException(
+                        'You can only specify one cluster environment '
+                        'Found more than 1 cluster environment plugin'
+                    )
+            else:
+                raise MisconfigurationException(
+                    f'Found invalid type for plugin {plug}. '
+                    'Expected a precision or training type plugin.'
+                )
+
+        self._training_type_plugin = training_type
+        self._training_type_plugin = self.training_type_plugin
+        self._precision_plugin = precision
+        self._cluster_environment = cluster_environment or self.select_cluster_environment()
+
+    @property
+    def precision_plugin(self) -> PrecisionPlugin:
+        if self._precision_plugin is None:
+            self._precision_plugin = self.select_precision_plugin()
+        return self._precision_plugin
+
+    @property
+    def training_type_plugin(self) -> TrainingTypePlugin:
+        if self._training_type_plugin is None:
+            self._training_type_plugin = self.select_training_type_plugin()
+        else:
+            self._training_type_plugin = self.resolve_training_type_plugin(self._training_type_plugin)
+
+        return self._training_type_plugin
+
+    @property
+    def cluster_environment(self) -> ClusterEnvironment:
+        return self._cluster_environment
+
     @property
     def on_cpu(self):
```

Comment:
Can we add types on all these properties?

Reply:
done in c0fbf7a